src/hooks/unstable_useSlots.ts

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+
+type SlotConfig = Record<string, React.ComponentType>
+
+type SlotElements<Type extends SlotConfig> = {
+  [Property in keyof Type]: React.ReactElement
+}
+
+/**
+ * Extract components from `children` so we can render them in different places,
+ * allowing us to implement components with SSR-compatible slot APIs.
+ * Note: We can only extract direct children, not nested ones.
+ */
+// eslint-disable-next-line camelcase
+export function unstable_useSlots<T extends SlotConfig>(
```

Comment:
Since this is internal-only, how would you feel about calling this `useSlots` instead of `unstable_useSlots`?

```suggestion
export function useSlots<T extends SlotConfig>(
```

We could then save `unstable_*` for exported members that are exempted from semver for the `npm` package.

Reply:
Done 👍